### ISSUE TITLE
Appropriate settings

### ISSUE THREAD

**Issue by lenlight** [Gazoo101/lyric-manager#1]
Hi Gazoo101, thanks for the great tool. In the USAGE EXAMPLE you state "set the appropriate settings, and execute main.py."

I seek to ask what you mean by appropriate settings.
I have renamed settings.yaml-example file to settings.yaml 

When I run main.py, I get an error
AttritubeError: 'NoneType' object has no attribute 'lower'

**Comment by Gazoo101** [comment]
I've added a check to more gracefully handle the issue so the error should now be more telling - thanks for the heads up @lenlight !

The missing parameter is likely the lyric_aligner option, which currently can either be NUSAutoLyrixAlignOffline or Disabled. The settings file has the options that are available. Note that if you choose the former you'll need to also provide the path to where you've installed NUSAutoLyrixAlignOffline.

**Comment by lenlight** [comment]
Thanks for Super Fast response. The error is the lyric aligner, just as you've said.
I have not locally installed NUSAutoLyrixAlign for Offline use, so it is disabled in my case.
Looking forward to the option for enabling Support of NUSAutoLyrixAlignOnline.

**Comment by Gazoo101** [comment]
My pleasure. I unfortunately balanced it out with a slower second response.

Unfortunately, the online version of NUSAutoLyrixAlign doesn't currently provide an API, meaning the only way to make it work via lyric-manager is to pretend to access the service as a browser. This is far more onerous work than having an API. In other words, it may be some time before that work gets done.

In the mean time, I recommend visiting the website and manually generating the text files for alignment (audacity version):

https://autolyrixalign.hltnus.org/

Then simply provide the generated .txt files and the lyric-manager can convert them as needed (and if needed).

**Comment by lenlight** [comment]
Thanks for the update. 
Will proceed to use the online version, meanwhile I will set up the offline version. which integrates well with the lyric-manager. What formats can the lyric-manager convert to?

**Comment by Gazoo101** [comment]
Currently a JSON file with timing information and supporting rendering detail. I've updated the readme.md on the repo to give much more detailed explanation of the input / output of Lyric Manager: https://github.com/Gazoo101/lyric-manager

Let me know if the details make sense?

**Comment by lenlight** [comment]
Thanks Gazoo101, have seen the updated readme.md
It's makes so much sense.

**Comment by Gazoo101** [comment]
My pleasure @lenlight ! If you run into any other issues - feel free to open another issue!

**Comment by lenlight** [comment]
Thanks Gazoo101, have one more query about the aligner, noted that when it
encounters a new word that does not exist in it's dictionary it replaces
the word with breathe*
Want to ask if one is able to add words in the dictionary to enable it to
recognize new words.

**Comment by Gazoo101** [comment]
My pleasure lenlight. You're probably better off asking this question directly to the authors of the aligner, here:

https://github.com/chitralekha18/AutoLyrixAlign

My experience with the aligner is that it tends to insert 'breathe*' when it cannot match a word it 'hears' in the audio, to a word in the lyrics. The lyric manager will attempt to manually match in these missing words, if it can reliably do so from the other aligned words.

In other words, the aligner may not be 'missing' the word, it just cannot reliably hear it. I've had some success with slightly altering words and getting the results I need. However, if the words you're referring to are indeed very custom, like names or something like 'Bebabobadoop' then yeah, it might impossible for it to 'hear' them without upgrading the underlying model.
